raft.go

```diff
@@ -927,6 +940,23 @@ func (r *Raft) appendEntries(rpc RPC, a *AppendEntriesRequest) {
 	return
 }
 
+// checkAndProcessConfigurationLog takes a log entry and updates the latest
+// configuration if the entry results in a new configuration. This must only be
+// called from the main thread.
```

Comment:
I haven't looked too deeply, but `checkAndProcessConfigurationLog()` is called from two places.  Does it make sense to put a guard in place w/ an atomic compare and swap op to make sure this isn't ever run concurrently?  If so, would that relax the restriction on this being called from only the main thread?